src/gui/AstroCalcDialog.cpp

```diff
@@ -3539,6 +3539,9 @@ void AstroCalcDialog::generateSolarEclipsesLocal()
 							treeItem->setData(SolarEclipseLocalDate, Qt::UserRole, JDmax);
 							treeItem->setText(SolarEclipseLocalType, eclipseTypeStr);
 							treeItem->setText(SolarEclipseLocalFirstContact, QString("%1").arg(localeMgr->getPrintableTimeLocal(JD1)));
+							if (centraleclipse)
```

Comment:
Why not use `if (centraleclipse && JD2<JD1)`?

Reply:
Yes, I'm about to change that. Thanks.